distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/config/ConfigureTestCommons.java

```diff
@@ -13,69 +13,101 @@
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.Dictionary;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
 import org.codice.ddf.configuration.DictionaryMap;
 import org.codice.ddf.itests.common.AdminConfig;
 import org.osgi.service.cm.Configuration;
 
 public class ConfigureTestCommons {
 
-  public static final String METACARD_VALIDITITY_FILTER_PLUGIN_SERVICE_PID =
+  private static final String METACARD_VALIDITY_FILTER_PLUGIN_SERVICE_PID =
       "ddf.catalog.metacard.validation.MetacardValidityFilterPlugin";
 
-  public static final String METACARD_VALIDITITY_MARKER_PLUGIN_SERVICE_PID =
+  private static final String METACARD_VALIDITY_MARKER_PLUGIN_SERVICE_PID =
       "ddf.catalog.metacard.validation.MetacardValidityMarkerPlugin";
 
-  public static final String METACARD_ATTRIBUTE_SECURITY_POLICY_PLUGIN_PID =
+  private static final String METACARD_ATTRIBUTE_SECURITY_POLICY_PLUGIN_PID =
       "org.codice.ddf.catalog.security.policy.metacard.MetacardAttributeSecurityPolicyPlugin";
 
-  public static final String AUTH_Z_REALM_PID = "ddf.security.pdp.realm.AuthzRealm";
+  private static final String AUTH_Z_REALM_PID = "ddf.security.pdp.realm.AuthzRealm";
 
   private ConfigureTestCommons() {}
 
-  public static void configureMetacardValidityFilterPlugin(
-      List<String> securityAttributeMappings, AdminConfig configAdmin) throws IOException {
-    Configuration config =
-        configAdmin.getConfiguration(METACARD_VALIDITITY_FILTER_PLUGIN_SERVICE_PID, null);
-    Dictionary<String, Object> properties = new DictionaryMap<>();
-    properties.put("attributeMap", securityAttributeMappings);
-    config.update(properties);
+  public static Dictionary<String, Object> configureService(
+      String pid, Dictionary<String, Object> props, AdminConfig adminConfig) throws IOException {
+    Configuration config = adminConfig.getConfiguration(pid, null);
+    Dictionary<String, Object> oldProps = config.getProperties();
+    config.update(props);
+
+    // verify update
+    with()
+        .pollInterval(1, SECONDS)
+        .await()
+        .atMost(TimeUnit.MINUTES.toSeconds(2), SECONDS)
+        .ignoreExceptions()
+        .until(
+            () -> {
+              Dictionary<String, Object> savedProps = config.getProperties();
+              List<String> newKeys = Collections.list(props.keys());
+              for (String key : newKeys) {
+                Object savedValue = savedProps.get(key);
+                Object incomingValue = props.get(key);
+                if (incomingValue.getClass().isArray() && savedValue.getClass().isArray()) {
+                  if (!Arrays.equals((Object[]) savedValue, (Object[]) incomingValue)) {
+                    return false;
+                  }
+                } else if (!savedValue.equals(incomingValue)) {
+                  return false;
+                }
+              }
+              return true;
+            });
+
+    return oldProps;
```

Comment:
✏️ It's a bit confusing to me that this has a return type or what the value of the return type would be without digging into this method. I would opt for this method not to return anything and have users retrieve the original configuration themselves before configuring. If you don't agree, we should at least javadoc that the original configuration before the update is being returned.

Reply:
I'm going javadoc route on this